### PR TITLE
Added linting with Lintly (flake8)

### PR DESCRIPTION
**Pull request by tankefugl**
This pull request adds a flake8 linting action to pull requests. This is configured to a maximum line length of 130, and only trigger on new issues (or, issues in changed code lines). This means that we can introduce it and improve the Python code over time.

Note that most style changes that will trigger flake8 can be easilly fixed by running autopep8, and this should therefore be a recommended practice.

I've added a Contributor section to the README.md including this.

I also added two changes that should trigger the linting, just to illustrate how it will capture and highlight this in the pull request.

Please note that the linter presently uses my account key to do the lint replies, this could be made by a bot user in the future.

### CODE REVIEW

File: pysilcam/tests/test_process.py [hunk 1]
```diff
@@ -38,8 +38,8 @@ def test_output_files():
     conf.write(conf_file_hand)
     conf_file_hand.close()
 
-    stats_file = os.path.join(data_file, 'proc', 'STN04-STATS.csv')
-    hdf_file = os.path.join(data_file, 'export/D20170509T172705.387171.h5')
+    stats_file = os.path.join(data_file,'proc', 'STN04-STATS.csv')
```
Comment (tankefugl) [inline pysilcam/tests/test_process.py:41]: E231: missing whitespace after ','

File: pysilcam/tests/test_process.py [hunk 2]
```diff
@@ -38,8 +38,8 @@ def test_output_files():
...
+    hdf_file = os.path.join(data_file,'export/D20170509T172705.387171.h5')
```
Comment (tankefugl) [inline pysilcam/tests/test_process.py:42]: E231: missing whitespace after ','

**Comment by emlynjdavies** [comment]
Looks very sensible, especially as it only triggers on new code.

Autopep8 is nice, but is has broken the code before, so must be checked by a human, especially is places not covered by the automated testing.

File: pysilcam/tests/test_process.py (hunk 1 above)
Comment (tankefugl) [inline pysilcam/tests/test_process.py:41]: E231: missing whitespace after ','

File: pysilcam/tests/test_process.py (hunk 2 above)
Comment (tankefugl) [inline pysilcam/tests/test_process.py:42]: E231: missing whitespace after ','

**Comment by tankefugl** [comment]
I've updated the merge rules to also require linting now.

File: pysilcam/tests/test_process.py (hunk 1 above)
Comment (tankefugl) [inline pysilcam/tests/test_process.py:41]: E231: missing whitespace after ','

File: pysilcam/tests/test_process.py (hunk 2 above)
Comment (tankefugl) [inline pysilcam/tests/test_process.py:42]: E231: missing whitespace after ','

**Comment by tankefugl** [comment]
@emlynjdavies Note that the builds of other branches now wait for linting because I turned it on! So let's either merge this one first, or disable then re-enable it later.

**Comment by emlynjdavies** [comment]
> I've updated the merge rules to also require linting now.

Ah. Ok, if it was just linting then maybe it’s not so critical I guess...? As long as this doesn’t happen for the tests too ;)

**Pull request merged**